Resolve rubocop offenses for auth_controller.rb

Eschon opened the pull request:
Since the last one was merged so fast I decided to do another file.
This time I'm not sure about how I fixed the long lines. Does anyone have a better idea?

NuckChorris commented on app/controllers/auth_controller.rb:
```diff
@@ -8,7 +8,10 @@ def sign_out_action
 
   def sign_in_action
-    return error! "Missing parameters", 400 unless [:email, :password].none? { |x| params[x].blank? }
+    return error! 'Missing parameters', 400 unless %i(email password)
```
Probably best to split it to a multi-line conditional

Eschon replied:
How would I do that? Won't rubocop complain when I change it to
```Ruby
unless [:email, :password].none? { |x| params[x].blank? }
  return error! 'Missing parameters', 400
end
```

NuckChorris replied:
I _hope_ it's smart enough to tell that it would violate another rule to flip it… but I don't know

NuckChorris commented on app/controllers/auth_controller.rb:
```diff
@@ -17,20 +20,25 @@ def sign_in_action
+                                                   .none? do |x|
+                                                     params[x].blank?
+                                                   end
+    return error! 'User with that email already exists', 409 if User.exists?(
```
These should probably get offloaded to model methods, but if you don't wanna bother right now this works well enough

Eschon commented:
I changed the long conditionals like @NuckChorris suggested and rubocop was ok with it.
I squashed everything so feel free to merge if nothing else needs to be changed.

NuckChorris commented:
:+1: You're really tearing through these! LGTM!

Eschon commented:
I don't have that much free time but sometimes I still don't know what to do with it :sweat_smile:
